src/components/CountryDetails.js

```diff
@@ -0,0 +1,59 @@
+import { Link, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import '../countries.json';
+import axios from 'axios';
+
+function CountryDetails(props) {
+
+  const { countryId } = useParams();
+  
+  console.log("this is country id ", {countryId})
+
+  const [details, setDetails] = useState({})
+
+
+  const { countries } = props;
+
+  useEffect(() => {
+      axios.get(`https://ih-countries-api.herokuapp.com/countries/${countryId}`)
+        .then((response) => {
+            setDetails(response.data)
+            console.log(response.data)
+        })
+        .catch(e => console.log("error getting country details", e));
+    }, [countryId])
+
+
+
+  return (
+    
+        <div>
+            
+            {details ? <p>Printing country details....</p> : <p>Loading details...</p>}
+
+            <p>Country: {details.name}</p>
+            <p>Country code: {details.alpha3Code}</p>
+            {/* <div>
+            <p>Borders:
+            {details.borders.map((code, item) => {
+              return(
+                <>
+                <li key={item}>
+                  <Link to={`/country/${code}`}>
+                    {
+                      countries.find((country) => country.alpha3Code === code)
+                        .name.official
+                    }
+                  </Link>
+                </li>
+                </>
+              )
+            })}</p> 
+            
+             </div> */}
```

Comment:
1. Since we are using bootstrap, you need to add the corresponding class to your div
2. You don't need to have a props to countries, because you are fetching data here already with axios. 
3. Getting props.countries wouldn't work anyways, because countries is not in the parents state, but in the sibling (CountriesList) 
4. When using map() method, you can add "?" to make avoid error if there are no borders

```
 return (
    <>
      <div className="col-7">
        {details && (
          <>
            <p>Country: {details.name.official}</p>
            <p>Country code: {details.alpha3Code}</p>
            <p>
              Borders:
              {details.borders?.map((code, item) => {
                return (
                  <>
                    <li key={item}>
                      <Link to={`/${code}`}>{code}</Link>
                    </li>
                  </>
                );
              })}
            </p>
          </>
        )}
      </div>
    </>
  );
  
```